server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1715,20 +1715,28 @@ public void verify(String seed, DiscoveryNode localNode) {
             }
         } else {
             BlobContainer testBlobContainer = blobStore().blobContainer(basePath().add(testBlobPrefix(seed)));
-            if (testBlobContainer.blobExists("master.dat")) {
-                try {
-                    BytesArray bytes = new BytesArray(seed);
-                    try (InputStream stream = bytes.streamInput()) {
-                        testBlobContainer.writeBlob("data-" + localNode.getId() + ".dat", stream, bytes.length(), true);
-                    }
-                } catch (IOException exp) {
-                    throw new RepositoryVerificationException(metadata.name(), "store location [" + blobStore() +
-                                                                               "] is not accessible on the node [" + localNode + "]", exp);
+            try {
+                BytesArray bytes = new BytesArray(seed);
+                try (InputStream stream = bytes.streamInput()) {
+                    testBlobContainer.writeBlob("data-" + localNode.getId() + ".dat", stream, bytes.length(), true);
                 }
-            } else {
-                throw new RepositoryVerificationException(metadata.name(), "a file written by master to the store [" + blobStore() + "] cannot be accessed on the node [" + localNode + "]. "
-                                                                           + "This might indicate that the store [" + blobStore() + "] is not shared between this node and the master node or "
-                                                                           + "that permissions on the store don't allow reading files written by the master node");
+            } catch (IOException exp) {
```

Comment:
Afaik ES catches `Exception` instead of `IOException` here - should we adapt this? Same further below (line 1738)

Reply:
Good point, lets do this.

Reply:
An the time of the commit from the backport it still catches `IOException`
https://github.com/elastic/elasticsearch/blob/8402ad95f2be603b7d939085723243750356f951/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java#L1699

Reply:
Ah later on they moved to Exception.